lib/TestScriptRunnable.rb

```diff
@@ -312,14 +353,16 @@ def successful? code
   end
 
   def storage(op)
-    self.reply = client.reply
-    reply.nil? ? return : client.reply = nil
+    self.reply = get_client(op.destination).reply
```

Comment:
Something to think about might be having a separate `client` method with an instance variable inside it, and you update that `@client` based off the operation when you first enter `execute_operation`. Might just be cleaner/avoid indexing every time client called. Something like:

```suggestion
    def client
         @client
    end
    
    set_client(operation.destination) # at the top of execute_operation, but instead of returning client, just set the @client instance
    
    self.reply = client.reply
```

Reply:
Great, looks much better!